### ISSUE TITLE
Dynamically turning on axis label does not resize plot

### ISSUE THREAD

**Issue by logmackenzie** [bokeh/bokeh#6748]
I am trying to hide the x-axis when my plot if fully zoomed out and then show it again as the user zooms in. Hiding it works fine and the plot automatically resizes to take up the extra space, but when I make the axis visible again, the plot does not resize to show the axis.

I am using Bokeh 0.12.6. I have been able to reproduce the problem with the following code: 

```
from bokeh.plotting import figure, show
import bokeh

plot = figure(title='Test Axis Visible', 
	tools='box_zoom, wheel_zoom, reset', 
	x_axis_label='X Axis', 
	y_axis_label='Y Axis',
	sizing_mode='stretch_both')

plot.rect([1, 2, 3, 4], [1, 2, 3, 4], height=1, width=1)

plot.xaxis.visible = False

plot.x_range = bokeh.models.Range1d(0, 5, bounds="auto")
plot.y_range = bokeh.models.Range1d(0, 5, bounds="auto")

plot.x_range.callback = bokeh.models.callbacks.CustomJS(args=dict(plot=plot), code="""
	var bounds = plot.x_range.bounds[1] - plot.x_range.bounds[0];
	var range = plot.x_range.max - plot.x_range.min;
	if (Boolean(range / bounds < 0.5)){
		plot.below[0].visible = true;
	} else {
		plot.below[0].visible = false;
	}
""")

show(plot)
```

When I zoom in on less than half the width of the figure, the x axis should appear, but I only see the cut off section shown below.

![axis bug](https://user-images.githubusercontent.com/30412735/29083112-a030a7a6-7c35-11e7-9be5-90100b9104a1.PNG)

**Comment by mattpap** [comment]
This is a known deficiency. You work around this issue by adding:
```
for (var id in Bokeh.index) {
    Bokeh.index[id].layout();
 }
```
at the end of callback's code. Note that this is a terrible hack.

**Comment by mattpap** [comment]
btw. This `if (Boolean(range / bounds < 0.5)){` doesn't need `Boolean()` to work as expected.

**Comment by logmackenzie** [comment]
Thanks!

**Comment by mattpap** [comment]
This issue is more or less a duplicate of other issues (e.g. #6676), but I will leave this one open for my own bookkeeping.

**Comment by bryevdv** [comment]
This appears to be fixed as of #8085 

![screenflow](https://user-images.githubusercontent.com/1078448/52168720-3549bc80-26e2-11e9-9791-62c7c715136e.gif)
